spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/config/StandardConfigDataLocationResolver.java

```diff
@@ -163,6 +165,27 @@ private Set<StandardConfigDataReference> getProfileSpecificReferences(ConfigData
 		return references;
 	}
 
+	private void validateProfiles(Profiles profiles) {
+		for (String profile : profiles) {
+			validateProfile(profile);
+		}
+	}
+
+	private void validateProfile(String profile) {
+		Assert.notNull(profile, "Profile must not be null");
```

Comment:
maybe we can combine null and empty checks in the single line with `Assert.state(!profile.isNullOrBlank())`

Reply:
Java has no such `isNullOrBlank` method, however, `Assert.hasText(...)` deals with `null` and empty strings.

Reply:
I removed `Assert.notNull()` and updated error message in `Assert.hasText()`.